app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/FahrplanFragment.java

```diff
@@ -334,13 +334,14 @@ private void addRoomColumns(
         adapterByRoomIndex.clear();
 
         int boxHeight = getNormalizedBoxHeight(getResources(), scale, LOG_TAG);
-        LecturesByRoomIndex lecturesByRoomIndex = new LecturesByRoomIndex(lectures);
         LayoutCalculator layoutCalculator = new LayoutCalculator(Logging.Companion.get(), boxHeight);
 
-        for (int roomIndex = 0; roomIndex < roomCount; roomIndex++) {
-            List<Lecture> roomLectures = lecturesByRoomIndex.get(roomIndex);
+        List<RoomData> roomDataList = scheduleData.getRoomDataList();
+        for (int roomIndex = 0; roomIndex < roomDataList.size(); roomIndex++) {
+            RoomData roomData = roomDataList.get(roomIndex);
```

Comment:
I appreciate the introduction of RoomData / ScheduleData. I've had something similar in mind already.
To me RoomData and ScheduleData are somewhat generic names though. What do you think about renaming them? 
ScheduleData => DaySchedule 
RoomData => Room
DaySchedule.getRoomDataList => DaySchedule.rooms
Room.roomName => Room.name